### ISSUE TITLE
split produces one million lines of error messages in the GNU test suite

### ISSUE THREAD

**Issue by jfinkels** [uutils/coreutils#4073]
The `tests/split/filter.sh` test is still producing about one million lines of `shell-init: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory`. So, maybe we should skip that part of the test for the moment? I believe it is this part:
```
# Ensure that "endless" input _is_ processed for unbounded number of filters    
for buf in 1000 1000000; do
  returns_ 124 timeout .5 sh -c \
    "yes | split --filter='head -c1 >/dev/null' -b $buf" || fail=1
done
```

_Originally posted by @jfinkels in https://github.com/uutils/coreutils/issues/3281#issuecomment-1288127837_

**Comment by sylvestre** [comment]
or do &> /dev/null to redirect stderr too :)

**Comment by jfinkels** [comment]
I believe this issue also causes this occasional failure:
```
GNU test failed: tests/misc/timeout. tests/misc/timeout is passing on 'main'. Maybe you have to rebase?
```

**Comment by tertsdiepraam** [comment]
Both options seem good to me. In the meantime, do we have any idea what's causing this? Seems like the working directory does not exist. How would something like that happen?

**Comment by tertsdiepraam** [comment]
I think @miles170 was right about this being fixed by #4315. Let's hope the timeout test is also no longer intermittent.

**Comment by jfinkels** [comment]
Excellent work @miles170, thank you!
